### PR TITLE
W-10619596: Aggregate notifications in mule operations

### PR DESCRIPTION
**Pull request by eze210**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/NotificationEmitterParser.java [hunk 1]
```diff
@@ -6,13 +6,17 @@
...
+import java.util.Optional;
+import java.util.function.Function;
 
 /**
  * @since 4.5.0
  */
 public interface NotificationEmitterParser {
 
-  List<String> getEmittedNotifications();
+  List<NotificationModel> getEmittedNotifications(Function<String, Optional<NotificationModel>> notificationMapper);
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/NotificationEmitterParser.java:20]: jdoc., specially around the mapper

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaSourceModelParser.java [hunk 2]
```diff
@@ -192,8 +194,10 @@ private Optional<Method> extractJavaMethod(Optional<MethodElement> method) {
   }
 
   @Override
-  public List<String> getEmittedNotifications() {
-    return NotificationModelParserUtils.getEmittedNotifications(sourceElement, getComponentTypeName(), getName());
+  public List<NotificationModel> getEmittedNotifications(Function<String, Optional<NotificationModel>> notificationMapper) {
+    List<String> identifiers =
+        NotificationModelParserUtils.getEmittedNotifications(sourceElement, getComponentTypeName(), getName());
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaSourceModelParser.java:199]: avoid unnecesary iterations and collections by having `NotificationModelParserUtils.getEmittedNotifications` delegate into an extracted `getEmmitedNotificationsStream` method.

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaSourceModelParser.java (hunk 2 above)
Comment (eze210) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaSourceModelParser.java:200]: done

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/NotificationEmitterParser.java (hunk 1 above)
Comment (eze210) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/NotificationEmitterParser.java]: done

**Comment by eze210** [comment]
--validate

**Comment by eze210** [comment]
--validate

**Comment by eze210** [comment]
--validate

**Review by marianogonzalez — APPROVED** [review]

**Pull request merged**